ironfish-rust/src/frost_utils/split_secret.rs

```diff
@@ -2,35 +2,44 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-use ironfish_frost::frost;
-use ironfish_frost::frost::Error;
 use ironfish_frost::frost::{
+    frost::keys::split,
     keys::{IdentifierList, KeyPackage, PublicKeyPackage},
     Identifier, SigningKey,
 };
 use rand::rngs::ThreadRng;
 use std::collections::HashMap;
 
+use crate::errors::{IronfishError, IronfishErrorKind};
+
 pub struct SecretShareConfig {
     pub min_signers: u16,
     pub max_signers: u16,
     pub secret: Vec<u8>,
 }
 
-pub fn split_secret(
+fn vec_to_array(vec: Vec<u8>) -> Result<[u8; 32], IronfishError> {
+    if vec.len() != 32 {
+        return Err(IronfishError::new(IronfishErrorKind::InvalidSecret));
+    }
+
+    let array: [u8; 32] = vec
+        .try_into()
+        .map_err(|_| IronfishError::new(IronfishErrorKind::InvalidSecret))?;
+
+    Ok(array)
+}
```

Comment:
If you provide a type parameter to the method and move to serializing this would be a generally useful method. I would return a generic serialization error from this method and then make it more refined where you call it in `split_key`

see this method for an example of the generic.
https://github.com/iron-fish/ironfish/blob/99763bbdc0238efff677fc11776c00d3e77f2d49/ironfish-rust/src/serializing/mod.rs#L50-L64

Reply:
As a matter of fact, this might already exist. In the code that you deleted on line 29, I think there may be an attempt to coerce the `vec` into an `array`, so this method won't be needed. You just will need to map the error to an ironfish error.

Reply:
Done!